src/components/input.tsx

```diff
@@ -0,0 +1,32 @@
+import { UseFormRegisterReturn } from "react-hook-form";
+import { twJoin } from "tailwind-merge";
+
+interface FormInputProps {
+  label: string;
+  register: UseFormRegisterReturn;
+  error?: string;
+  className?: string;
+  placeholder?: string;
+}
+
+const INPUT_CLASSES = "border border-gray-300 p-2 rounded";
+
+export function Input(props: FormInputProps) {
+  const { label, register, error, className, placeholder } = props;
+
+  const inputWithExternalClasses = twJoin(INPUT_CLASSES, className);
+
+  return (
+    <div className="mb-4">
+      <div className="flex flex-row items-center gap-4 justify-between">
+        <label className="block">{label}:</label>
+        <input
+          {...register}
+          className={inputWithExternalClasses}
+          placeholder={placeholder}
+        />
+      </div>
+      {error && <span className="text-red-500 font-bold">{error}</span>}
```

Comment:
Could you please add a variable to evaluate if "error" was passed to the component and replace the short circuit sintax to a full ternary?
Using short circuit sintax can lead to potential errors that could be easily avoided using ternary. 
As reference:
https://medium.com/geekculture/stop-using-for-conditional-rendering-in-react-a0f7b96200f8
https://www.showrin.com/blog/the-myth-you-should-know-about-short-circuit-syntax-in-jsx
```suggestion
      // outside the HTML part
      const hasError = error != null
      
      // replacement
      {hasError ? <span className="text-red-500 font-bold">{error}</span> : null}
```

Reply:
didnt know about that, good to know and i dont thing that im gonna use that ever again, just to avoid bugs